src/utils.py

```diff
@@ -0,0 +1,102 @@
+    """
+    Kiírja a menüt a megadott opciókkal majd visszaadja a felhasználó választását.
+
+    Args:
+    options(dict[int, str]): Menü opciók, ahol a kulcsok a menüpont sorszámai, az értékek pedig a nevek.
+
+    Returns:
+    int: A felhasználó által választott menüpont sorszáma.
+    """
+    
+    while True:
+        print("--------------------")
+        for number,title in options.items():
+            print(f"{number} - {title}")
+        print("--------------------")
+        try:
+            choice = int(input("Válasszon menüpontot: "))
+        except ValueError:
+            print("HIBA! Csak számot adhat meg!")
+            print()
+            continue
+            
+        if choice in options: return choice
+        else: print("HIBA! Nincs ilyen menüpont!"); print()
+
+def load_json(filename: str) -> dict:
+    """
+    Beolvassa a megadottt json filet, majd dict-ként visszaadja a tartalmát.
+    
+    Args:
+    file (str): A JSON file neve.
+    
+    Returns:
+    dict: A fájl tartalma dict-ként.
+    
+    Raises:
+    FileNotFoundError:  Ha nem található a file.
+    JSONDecodeError: Ha hibás JSON formátumú a file.
+    """
+    try:
+        with open(filename, 'r', encoding='utf-8') as f:
+            return json.load(f)            
+    except FileNotFoundError as e:
+          raise FileNotFoundError(f"Nem található a '{filename}' fájl.") from e
+    except json.JSONDecodeError as e:
+        raise json.JSONDecodeError(f"Hibás JSON formátum '{filename}' fájlban.") from e
+
+def save_json(data: dict, filename: str) -> None:
+    """
+    Elmenti az adatokat a megadott JSON fileba.
+    
+    Args:
+    data (dict): Az elmenteni kívánt adatok.
+    filename (str): A kimeneti fájl.
+    
+    Raises:
+    IOError: Ha hiba történik íráskor.
+    """
+    
+    try:
+        with open(filename,"w") as f:
+            json.dump(data, f, ensure_ascii=False, indent=4)
+    except IOError as e:
+        raise IOError(f"Hiba a {filename} fájl mentése közben.") from e
+    
+def list_names_with_ids(data: dict) -> None:
+    """
+    Kilistázza a dictben található elemek nevét és azonosítójá
+
+    Args:
+    data (list[dict]): A dictben található elemek.
+
+    Raises:
+    ValueError: Ha a dict üres.
+    """
+    
+    if not data:
+        raise ValueError("A lista üres.")
+    
+    for i,resident in enumerate(data):
+        print(f"{i+1}. {resident['name']}  ({resident['id']})")
+        
+def filter_list_by_ids(datas:list[dict],selected_ids:list[dict]) -> list[dict]:
+    """
+    Visszaadja azokat az adatokat, melyek id-ja benne van a beviteli változóban.
+    
+    Args:
+    datas: Az összes adatot tartalmazó lista.
+    selected_ids: Azok az id-k amelyekre szűrni szeretnénk.
+    
+    Returns:
+    list[dict]: A szűrt adatok listája.
+    """
+    filtered_datas = []
+    for data in datas:
+        if data['id'] in selected_ids:
+            filtered_datas.append(data)
+    return filtered_datas
```

Comment:
Ezt szépen meg lehet oldani egy list comprehensionnel 1 sorban:
```suggestion
    return [x for x in datas if x['id'] in selected_ids]
```